App/Redux/TextileRedux.js

```diff
@@ -67,18 +83,28 @@ export const handleImageProgress = (state, {data}) => {
 }
 
 export const handleImageUploadComplete = (state, {data}) => {
-  const {file} = data
-  // Remove the item from Redux storage
-  const items = state.images.items.filter(item => item.remotePayloadPath !== file)
+  const { uploadId } = data
+  const existingItems = state.images.items ? state.images.items : []
+  const items = existingItems.map(item => {
+    if (item.uploadId === uploadId) {
+      return {...item, state: 'cleanup'}
```

Comment:
Why is this additional `cleanup` state necessary?

Reply:
not necessary, i just figured if we get around to completing all the error handling, there is the potential error of an image uploading correctly but failing to cleanup it's multipart data